pkg/serviceaccounttoken/secret_test.go

```diff
@@ -340,10 +343,212 @@ func TestServiceAccountSecret(t *testing.T) {
 }
 
 type fakeSecretLister struct {
-	secrets []*v1.Secret
+	secrets []*corev1.Secret
 	err     error
 }
 
-func (f *fakeSecretLister) list(namespace string, selector labels.Selector) ([]*v1.Secret, error) {
+func (f *fakeSecretLister) list(namespace string, selector labels.Selector) ([]*corev1.Secret, error) {
 	return f.secrets, f.err
 }
+
+func TestEnsureSecretForServiceAccount_in_parallel(t *testing.T) {
+	managedSecrets := map[string]*corev1.Secret{
+		"test-secret-1": &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-secret-1",
+				Namespace: "default",
+				Labels: map[string]string{
+					ServiceAccountSecretLabel: "test",
+				},
+			},
+			Type: corev1.SecretTypeOpaque,
+		},
+		"test-secret-2": &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-secret-2",
+				Namespace: "default",
+				Labels: map[string]string{
+					ServiceAccountSecretLabel: "test",
+				},
+			},
+			Type: corev1.SecretTypeOpaque,
+		},
+	}
+
+	k8sClient := fake.NewSimpleClientset()
+	var m sync.Mutex
+
+	k8sClient.PrependReactor("list", "secrets",
+		func(a k8stesting.Action) (bool, runtime.Object, error) {
+			m.Lock()
+			defer m.Unlock()
+
+			secrets := &corev1.SecretList{}
+			for _, v := range managedSecrets {
+				secrets.Items = append(secrets.Items, *v)
+			}
+
+			return true, secrets, nil
+		},
+	)
+
+	k8sClient.PrependReactor("create", "secrets",
+		func(action k8stesting.Action) (bool, runtime.Object, error) {
+			m.Lock()
+			defer m.Unlock()
+			ret := action.(k8stesting.CreateAction).GetObject().(*corev1.Secret)
+			ret.ObjectMeta.Name = ret.GenerateName + rand.String(5)
+			ret.ObjectMeta.Labels = map[string]string{
+				ServiceAccountSecretLabel: "test", // sa.Name from below
```

Comment:
Is this necessary? I would have thought that the code being tested should be adding this when it makes the create call (and that this should be something that we can look for/assert on).